Azure-Samples/azure-search-openai-demo
# Enable uploads of large PDFs over connections with less upload speed

## Purpose

The Azure Python SDK defaults to a "max put size" of 64 MB, meaning that they'll attempt to upload up to 64 MB within the default timeout period of 20 seconds. Any blob over 64 MB will be chunked up.

Unfortunately, on my home connection, I cannot upload even 50 MB within that timeout period, and I've seen other developers comment in the SDK repo with similar issues. So this change configures single_max_put_size to 4 MB for the blob manager, which means it'll upload up to 4 MB at a time, and chunk after that. I was able to upload a 50 MB file with this change.

CC @jalauzon-msft who advised me on the change and commented on this related issue: https://github.com/Azure/azure-sdk-for-python/issues/23232

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Put 50 MB PDF into data (like https://www.el-cerrito.org/DocumentCenter/View/5415/UrbanGreeningPlan_Final-Draft-121515?bidId=)
* Run ./scripts/prepdoc.sh
* It should succeed
